tests/components/xiaomi_aqara/test_config_flow.py

```diff
@@ -0,0 +1,324 @@
+"""Test the Xiaomi Aqara config flow."""
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.components.xiaomi_aqara import config_flow, const
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT
+
+from tests.async_mock import Mock, patch
+
+ZEROCONF_NAME = "name"
+ZEROCONF_PROP = "properties"
+ZEROCONF_MAC = "mac"
+
+TEST_HOST = "1.2.3.4"
+TEST_HOST_2 = "5.6.7.8"
+TEST_KEY = "1234567890123456"
+TEST_PORT = 1234
+TEST_NAME = "Test_Aqara_Gateway"
+TEST_SID = "abcdefghijkl"
+TEST_PROTECOL = "1.1.1"
+TEST_MAC = "ab:cd:ef:gh:ij:kl"
+TEST_GATEWAY_ID = TEST_MAC
+TEST_ZEROCONF_NAME = "lumi-gateway-v3_miio12345678._miio._udp.local."
+
+
+@pytest.fixture(name="xiaomi_aqara", autouse=True)
+def xiaomi_aqara_fixture():
+    """Mock xiaomi_aqara discovery and entry setup."""
+    mock_gateway_discovery = get_mock_discovery([TEST_HOST])
+
+    with patch(
+        "homeassistant.components.xiaomi_aqara.config_flow.XiaomiGatewayDiscovery",
+        return_value=mock_gateway_discovery,
+    ), patch(
+        "homeassistant.components.xiaomi_aqara.config_flow.get_mac_address",
+        return_value=TEST_MAC,
+    ), patch(
+        "homeassistant.components.xiaomi_aqara.async_setup_entry", return_value=True
+    ):
+        yield
+
+
+def get_mock_discovery(host_list):
+    """Return a mock gateway info instance."""
+    gateway_discovery = Mock()
+
+    gateway_dict = {}
+    for host in host_list:
+        gateway = Mock()
+
+        gateway.ip_adress = host
+        gateway.port = TEST_PORT
+        gateway.sid = TEST_SID
+        gateway.proto = TEST_PROTECOL
+
+        gateway_dict[host] = gateway
+
+    gateway_discovery.gateways = gateway_dict
+
+    return gateway_discovery
+
+
+async def test_config_flow_user_success(hass):
+    """Test a successful config flow initialized by the user."""
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We need to patch the calls to the library so no I/O is done in the tests.

Reply:
I thought the pytest.fixture already took care of that (line 27-41)?:
```
@pytest.fixture(name="xiaomi_aqara", autouse=True)
def xiaomi_aqara_fixture():
    """Mock xiaomi_aqara discovery and entry setup."""
    mock_gateway_discovery = get_mock_discovery([TEST_HOST])

    with patch(
        "homeassistant.components.xiaomi_aqara.config_flow.XiaomiGatewayDiscovery",
        return_value=mock_gateway_discovery,
    ), patch(
        "homeassistant.components.xiaomi_aqara.config_flow.get_mac_address",
        return_value=TEST_MAC,
    ), patch(
        "homeassistant.components.xiaomi_aqara.async_setup_entry", return_value=True
    ):
        yield
```

Reply:
Right. I missed that fixture. :+1: